sunscreen_runtime/src/debugger/server.rs

```diff
@@ -284,16 +294,34 @@ pub async fn get_stack_trace(
     let sessions = get_sessions().lock().unwrap();
 
     if let Some(curr_session) = sessions.get(&session) {
-        let curr_session = curr_session.unwrap_bfv_session();
-        let stack_lookup = &curr_session.graph.metadata.stack_lookup;
-
-        if let Some(node_info) = curr_session.graph.node_weight(NodeIndex::new(nodeid)) {
-            if let Some(stack_frames) = stack_lookup.id_data_lookup.get(&node_info.stack_id) {
-                let stack_frames_json = serde_json::to_string(&stack_frames).unwrap();
-                return Ok(HttpResponse::Ok().body(stack_frames_json));
-            } else {
-                return Ok(HttpResponse::NotFound()
-                    .body(format!("Stack trace for node {} not found", nodeid)));
+        match curr_session {
+            Session::BfvSession(curr_session) => {
+                let stack_lookup = &curr_session.graph.metadata.stack_lookup;
+
+                if let Some(node_info) = curr_session.graph.node_weight(NodeIndex::new(nodeid)) {
+                    if let Some(stack_frames) = stack_lookup.id_data_lookup.get(&node_info.stack_id)
+                    {
+                        let stack_frames_json = serde_json::to_string(&stack_frames).unwrap();
+                        return Ok(HttpResponse::Ok().body(stack_frames_json));
+                    } else {
+                        return Ok(HttpResponse::NotFound()
+                            .body(format!("Stack trace for node {} not found", nodeid)));
+                    }
+                }
+            }
+            Session::ZkpSession(curr_session) => {
+                let stack_lookup = &curr_session.graph.metadata.stack_lookup;
+
+                if let Some(node_info) = curr_session.graph.node_weight(NodeIndex::new(nodeid)) {
+                    if let Some(stack_frames) = stack_lookup.id_data_lookup.get(&node_info.stack_id)
+                    {
+                        let stack_frames_json = serde_json::to_string(&stack_frames).unwrap();
+                        return Ok(HttpResponse::Ok().body(stack_frames_json));
+                    } else {
+                        return Ok(HttpResponse::NotFound()
+                            .body(format!("Stack trace for node {} not found", nodeid)));
+                    }
+                }
```

Comment:
You could clean all this up by having a method `get_stack_trace(&self, id: NodeIndex) -> Option<StackFrames>` on the session. Instead of nesting all these if statements you can just use ? to propagate the Option in there.